### PR TITLE
Limit bucket count in selection

### LINKED ISSUE #6078: Max group by buckets - Prevent queries that would create too many buckets (e.g. group by time(1s) over 1y of points)
#6024

### PR DESCRIPTION
**Pull request by jsternberg**
Fixes #6078.

### CODE REVIEW

File: cluster/query_executor.go [hunk 1]
```diff
@@ -469,6 +470,25 @@ func (e *QueryExecutor) executeSelectStatement(stmt *influxql.SelectStatement, c
...
+	if e.MaxSelectBucketsN > 0 && !stmt.IsRawQuery {
+		interval, err := stmt.GroupByInterval()
+		if err != nil {
+			return err
+		}
+
+		if interval > 0 {
+			// Determine the start and end time matched to the interval (may not match the actual times).
+			min := opt.MinTime.Truncate(interval)
+			max := opt.MaxTime.Truncate(interval).Add(interval)
```
Comment (e-dard) [inline cluster/query_executor.go]: small nit: `max := min.Add(interval)`

**Comment by e-dard** [comment]
LGTM :+1:

**Comment by jwilder** [comment]
:+1:

File: cluster/query_executor.go (hunk 1 above)
Comment (jsternberg) [inline cluster/query_executor.go]: Haha, can do. :+1:

**Pull request merged**